src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc

```diff
@@ -452,12 +452,15 @@ void PX4FirmwarePlugin::guidedModeChangeAltitude(Vehicle* vehicle, double altitu
 
 void PX4FirmwarePlugin::startMission(Vehicle* vehicle)
 {
-    if (!_armVehicleAndValidate(vehicle)) {
-        qgcApp()->showMessage(tr("Unable to start mission: Vehicle failed to arm."));
-        return;
-    }
 
-    _setFlightModeAndValidate(vehicle, missionFlightMode());
+    if (_setFlightModeAndValidate(vehicle, missionFlightMode())) {
```

Comment:
The problem is that with my testing (mission with waypoint too far) setting the flight mode doesn't fail. See discussion in #5415